be/src/exprs/time_functions.cpp

```diff
@@ -2106,4 +2106,41 @@ std::string TimeFunctions::info_reported_by_time_slice = "time used with time_sl
 #undef DEFINE_TIME_STRING_UNARY_FN
 #undef DEFINE_TIME_UNARY_FN_EXTEND
 
+static int weekday_from_dow_abbreviation(const std::string& dow) {
+    const int err_tag = -1, base = 1000;
+    if (dow.length() < 2) {
+        return err_tag;
+    }
+    switch (dow[0] + dow[1] * base) {
+    case 'S' + 'u' * base:
+        return (dow == "Su" || dow == "Sun" || dow == "Sunday") ? 0 : err_tag;
+    case 'M' + 'o' * base:
+        return (dow == "Mo" || dow == "Mon" || dow == "Monday") ? 1 : err_tag;
+    case 'T' + 'u' * base:
+        return (dow == "Tu" || dow == "Tue" || dow == "Tuesday") ? 2 : err_tag;
+    case 'W' + 'e' * base:
+        return (dow == "We" || dow == "Wed" || dow == "Wednesday") ? 3 : err_tag;
+    case 'T' + 'h' * base:
+        return (dow == "Th" || dow == "Thu" || dow == "Thursday") ? 4 : err_tag;
+    case 'F' + 'r' * base:
+        return (dow == "Fr" || dow == "Fri" || dow == "Friday") ? 5 : err_tag;
+    case 'S' + 'a' * base:
+        return (dow == "Sa" || dow == "Sat" || dow == "Saturday") ? 6 : err_tag;
+    default:
+        return err_tag;
+    }
+}
+
+// next_day
+DEFINE_BINARY_FUNCTION_WITH_IMPL(next_dayImpl, datetime, dow) {
+    int datetime_weekday = ((DateValue)datetime).weekday();
+    int dow_weekday = weekday_from_dow_abbreviation(dow.to_string());
+    if (dow_weekday != -1) {
+        return (DateValue)timestamp_add<TimeUnit::DAY>(datetime, (6 + dow_weekday - datetime_weekday) % 7 + 1);
+    } else {
+        throw std::runtime_error(dow.to_string() + " not supported in next_day dow_string");
+    }
+}
+DEFINE_TIME_CALC_FN(next_day, TYPE_DATETIME, TYPE_VARCHAR, TYPE_DATE);
```

Comment:
I think we can add a prepare function for next_day, so that we can avoid call `weekday_from_dow_abbreviation `once per row if the second parameter is constant string.

Reply:
you could leverage the prepare/close function to handle constant parameter

Reply:
Nice suggestion! I have finished it.